front-end/src/components/MatrixInput.tsx

```diff
@@ -0,0 +1,62 @@
+import { ReactElement } from "react"
+import { MatrixInputProps } from "@shared/utils/matrixInput.ts"
+import { Markdown } from "@/components/Markdown.tsx"
+import { useTheme } from "@/hooks/useTheme.ts"
+
+/**
+ * Create the parentheses for the matrix input
+ * @param svgColor - either "white" or "black" depending on the user's theme
+ * @param side - "l" --> left parentheses of the matrix | "r" follows
+ */
+function Parentheses({ svgColor, side }: { svgColor: string; side: "r" | "l" }) {
+  const svgPath =
+    side === "l"
+      ? "M85 0 A61 101 0 0 0 85 186 L75 186 A65 101 0 0 1 75 0"
+      : "M24 0 A61 101 0 0 1 24 186 L34 186 A65 101 0 0 0 34 0"
+  return (
+    <div className="h-full w-[0.85em]">
+      <svg width="100%" height="100%" preserveAspectRatio="none" viewBox="3 0 106 186">
+        <path d={svgPath} fill={svgColor} stroke={svgColor}></path>
+      </svg>
+    </div>
+  )
+}
+
+/**
+ * Creates a component for matrix input
+ * @param matrixObject
+ */
+export function MatrixInput({ matrixObject }: { matrixObject: string }): ReactElement {
+  const parsedMatrixObject = JSON.parse(matrixObject) as MatrixInputProps
+  const rows = parsedMatrixObject.rows
+  const cols = parsedMatrixObject.cols
+
+  const { theme } = useTheme()
+
+  const divHeight = 80 + (rows - 2) * 45
```

Comment:
These magic numbers depend on the height of the input fields. I assume they are inferred by trial and error. At least I couldn't find the height definitions of the input fields or they are defined as `h-8` or something like that. Maybe there is a good html/css solution?

Reply:
I took a look again and using the web inspector it seems like each field (with padding) is 42px. So I would change this to `const divHeight = rows * 42`. 
tested it for different values from 1 to 22 and looks good for those. Its still a _magic number_ but not trail and error anymore